common.py

```diff
@@ -2,33 +2,62 @@
 import os
 import time
 import json
+import logging
+import subprocess
 
 class CommonWebPageFuncs:
     def __init__(self):
-        #Could these come from a config? Also, inconsistent naming scheme. Please change.
-        self.ValidationPath = os.getenv("ValidationPath")
-        self.GIT_PULL_REQUEST = os.getenv("GIT_PULL_REQUEST", "false")
-        self.GIT_PULL_REQUEST_TITLE = os.getenv("GIT_PULL_REQUEST_TITLE", "")
-        self.GIT_PULL_REQUEST_LINK = os.getenv("GIT_PULL_REQUEST_LINK", "")
-        self.GIT_COMMIT = os.getenv("GIT_COMMIT", "")
-        self.GIT_TOKEN = os.getenv("GitHubToken", "")
-
-        #Some variables to read from a config file - this is set up as an example. May want to expand this later.
-        with open(f"{self.ValidationPath}/git_setup.json") as config_file: #THis config file can be passed as an argument to the class init, do we want to do this?
-            data = json.load(config_file)
-
-        #Git config
-        self.GIT_USER = data["Commit"]["Username"]
-        self.GIT_EMAIL = data["Commit"]["Email"]
-        #Validation setup
-        self.WEBPAGE_BRANCH = data["Validation"]["WebPageBranch"]
-        self.WEBPAGE_FOLDER = data["Validation"]["WebPageFolder"]
-        self.VALIDATION_GIT_PATH = data["Validation"]["Path"]
-        self.VALIDATION_GIT_BRANCH = data["Validation"]["CodeBranch"]
-        #Software setup
-        self.SOFTWARE_NAME = data["Software"]["Name"]
-        self.SOFTWARE_GIT_PATH = data["Software"]["Path"]
-        
+
+        #Set up logging. A more useful print that allows us to define errors, warnings, normal messages and what level these pintouts happen.
+        #i.e. Since this script is used as an import to the run scripts, the logger will report that the error happened in Common.

```

Comment:
That was indeed the hope! In truth I was slightly over ambitious with the logging, it may take a PR on it's own in order to produce useful logging (more useful than just the print statements we use at the moment!). I will add an issue for logging to be worked on in the future